### PR TITLE
HDDS-6751. Migrate parameterized tests in hdds-server-scm to JUnit5

### PR DESCRIPTION
**Pull request by kaijchen**
## What changes were proposed in this pull request?

Migrate tests with `@RunWith(Parameterized.class)` in hdds-server-scm to JUnit5

```
hadoop-hdds/server-scm/src/test/java/org/apache/hadoop/hdds/scm/TestStorageContainerManagerHttpServer.java
hadoop-hdds/server-scm/src/test/java/org/apache/hadoop/hdds/scm/container/placement/algorithms/TestSCMContainerPlacementRackAware.java
hadoop-hdds/server-scm/src/test/java/org/apache/hadoop/hdds/scm/container/placement/algorithms/TestSCMContainerPlacementRackScatter.java
hadoop-hdds/server-scm/src/test/java/org/apache/hadoop/hdds/scm/pipeline/TestPipelineDatanodesIntersection.java
hadoop-hdds/server-scm/src/test/java/org/apache/hadoop/hdds/scm/upgrade/TestSCMHAUnfinalizedStateValidationAction.java
```

~~In addition, one more test in hdds-server-scm was changed to `@ParameterizedTest`.~~
~~If there is more, they can be changed in HDDS-6752 after this PR is merged.~~
~~`hadoop-hdds/server-scm/src/test/java/org/apache/hadoop/hdds/scm/safemode/TestSCMSafeModeManager.java`~~

EDIT: We should probably do this in another JIRA.

## What is the link to the Apache JIRA

https://issues.apache.org/jira/browse/HDDS-6751

## How was this patch tested?

CI: https://github.com/kaijchen/ozone/actions/runs/2343964748

### CODE REVIEW

File: hadoop-hdds/server-scm/src/test/java/org/apache/hadoop/hdds/scm/pipeline/TestPipelineDatanodesIntersection.java [hunk 1]
```diff
@@ -127,7 +127,7 @@ public void testPipelineDatanodesIntersection(int nodeCount,
         List<Pipeline> overlapPipelines = RatisPipelineUtils
             .checkPipelineContainSameDatanodes(stateManager, pipeline);
 
-        if (overlapPipelines.isEmpty()) {
```
Comment (kaijchen) [inline hadoop-hdds/server-scm/src/test/java/org/apache/hadoop/hdds/scm/pipeline/TestPipelineDatanodesIntersection.java:144]: I believe this if condition needs to be fixed, see line 132.

Comment (kaijchen) [reply hadoop-hdds/server-scm/src/test/java/org/apache/hadoop/hdds/scm/pipeline/TestPipelineDatanodesIntersection.java:144]: However, there is no assertions against any of these data.
We should fix this test probably in another Jira.

File: hadoop-hdds/server-scm/pom.xml [hunk 2]
```diff
@@ -150,6 +150,10 @@ https://maven.apache.org/xsd/maven-4.0.0.xsd">
       <artifactId>junit-platform-launcher</artifactId>
       <scope>test</scope>
     </dependency>
+    <dependency>
+      <groupId>org.junit.jupiter</groupId>
+      <artifactId>junit-jupiter-params</artifactId>
```
Comment (kaijchen) [inline hadoop-hdds/server-scm/pom.xml:155]:
```suggestion
      <artifactId>junit-jupiter-params</artifactId>
      <scope>test</scope>
```

**Review by adoroszlai — APPROVED** [review]
Thanks @kaijchen for the patch, LGTM.

**Comment by kaijchen** [comment]
Thanks @adoroszlai for the review!

**Pull request merged**